src/core/networkreply.cpp

```diff
@@ -41,10 +41,7 @@ void NetworkReply::abort()
 
 QNetworkReply *NetworkReply::reply() const
 {
-  if ( mIsFinished )
-    return mReply;
-
-  return nullptr;
+  return mReply;
```

Comment:
@suricactus , was there a reason why you were not passing on the pointer until finished?

Reply:
Yes, because there is the internal retry and the `QNetworkReply` object might change. if this is going to be changed, please also update the docs here: https://github.com/opengisch/QField/blob/QField-Cloud-Projects/src/core/networkreply.h#L61

Reply:
Ok we'll update the documentation then. We definitely need access to it during download process so we offload bits from memory to disk as they come in.

Reply:
Be careful that the `QNetworkReply` might be deleted even if the parent `NetworkReply` is still alive.